pkg/storage/db.go

```diff
@@ -866,23 +882,107 @@ func (db *DB) releasePublicIP(ctx context.Context, tx *sql.Tx, arnID string, ip
 	//to NOT have AWS resources by definition
 	//this way we can find unbalanced events while avoiding nullable not_before
 	//which provides minimal support for out-of-order events
-	const releasePublicIPQuery = `
-do
-$$
-    begin
+	const releasePublicIPQueryUpdate = `
         update aws_public_ip_assignment
         set not_after=$1
         where public_ip = $2
           and aws_resource_id = (select id from aws_resource where arn_id = $3)
-          and aws_hostname = $4;
-        if not found then
+          and aws_hostname = $4`
```

Comment:
is it possible for there to be multiple records with the same public_ip, aws_resource_id, and aws_hostname (but different not_before/not_after values)? If so, might need a "where" clause to add "and not_after is null" so that the update doesn't overwrite all not_after values of other records

Reply:
I feel like aws_resource_id's should be unique 🤔

Reply:
Very good point, I will look into this.

Reply:
Done.